src/core/ReactCompositeComponent.js

```diff
@@ -485,7 +494,7 @@ var ReactCompositeComponentMixin = assign({},
    */
   _maskContext: function(context) {
     var maskedContext = null;
-    var contextTypes = this._instance.constructor.contextTypes;
+    var contextTypes = this._currentElement.type.contextTypes;
```

Comment:
This is kind of a perf optimization since this lookup happens in production right now. We never need contextTypes for wrappers right now so this allows us to by-pass the class lookup. Not really a solid solution. I could add a TODO.